### PR TITLE
Change use of NodeItem to AddrPort in node base classes

### PR DESCRIPTION
**Pull request by gavin-norman-sociomantic**

### CODE REVIEW

**Comment by gavin-norman-sociomantic** [comment]
Note to reviewer: could you confirm the changes around `InetAddress` and `AddrPort`? It's quite complicated.

**Comment by gavin-norman-sociomantic** [comment]
(I was originally intending to replace all internal usage of `NodeItem` with `AddrPort` -- see https://github.com/sociomantic-tsunami/swarm/issues/348 -- but this is a massive undertaking, as it's used extensively in the client. This PR just updates a small section -- the node bases.)

File: src/swarm/node/model/INodeInfo.d [hunk 1]
```diff
@@ -38,9 +39,20 @@ public interface INodeInfo
 
      **************************************************************************/
 
+    deprecated("Use addr_port instead")
     public NodeItem node_item ( );
```
Comment (nemanja-boric-sociomantic) [inline src/swarm/node/model/INodeInfo.d:43]: Deprecation node?

File: src/swarm/node/model/INodeInfo.d [hunk 2]
```diff
@@ -38,9 +39,20 @@ public interface INodeInfo
...
 
 
+    /**************************************************************************
+
+        Returns:
+            NodeItem struct containing the address & port of the node
+
+     **************************************************************************/
+
+    public AddrPort addr_port ( );
```
Comment (nemanja-boric-sociomantic) [inline src/swarm/node/model/INodeInfo.d:53]: Breaking change?

**Comment by gavin-norman-sociomantic** [comment]
Blocked pending testing with proto and node repos.

File: src/swarm/node/model/NeoNode.d [hunk 3]
```diff
@@ -845,13 +891,18 @@ public class NodeBase ( ConnHandler : ISwarmConnectionHandler ) : INodeBase
         }
 
         // Super ctor.
+        InetAddress!(false) addr, neo_addr;
+        addr = cast(sockaddr_in)node;
```
Comment (nemanja-boric-sociomantic) [inline src/swarm/node/model/NeoNode.d:896]: Consider adding `opCast` from `AddrPort` directly to `InetAddr`.

File: src/swarm/node/model/NeoNode.d [hunk 4]
```diff
@@ -845,13 +891,18 @@ public class NodeBase ( ConnHandler : ISwarmConnectionHandler ) : INodeBase
...
+        neo_addr = addr;
+        neo_addr.port = neo_port;
+
         super(node, conn_setup_params,
             this.listener = new Listener(
-                addr(node.Address, node.Port), this.socket, conn_setup_params,
+                cast(sockaddr*)&addr.addr, this.socket, conn_setup_params,
```
Comment (nemanja-boric-sociomantic) [inline src/swarm/node/model/NeoNode.d:902]: This is fine, considering that `addr.addr` is `sockaddr_in`. However, I think by casting pointer we're loosing the type safety here. I would just add the static assert before `static assert(typeof(addr.addr) is sockaddr_in)`.

File: src/swarm/node/model/INodeInfo.d (hunk 1 above)
Comment (gavin-norman-sociomantic) [reply src/swarm/node/model/INodeInfo.d:43]: Could you elaborate?

**Comment by gavin-norman-sociomantic** [comment]
On second thoughts, I'd prefer to not address this now. We plan to completely rewrite all of the core node stuff, anyway, at some point, so it seems better to leave it alone until then.

Replaced with #354, #355.

**Pull request closed without merging**